platform/mbed_mem_trace.c

```diff
@@ -26,10 +26,13 @@
 
 /* The callback function that will be called after a traced memory operations finishes. */
 static mbed_mem_trace_cb_t mem_trace_cb;
-/* 'trave_level' guards "trace inside trace" situations (for example, the implementation
+/* 'trace_lock_count' guards "trace inside trace" situations (for example, the implementation
  * of realloc() might call malloc() internally, and since malloc() is also traced, this could
  * result in two calls to the callback function instead of one. */
-static uint8_t trace_level;
+static uint8_t trace_lock_count;
+
+#define TRACE_IS_LOCKED() (trace_lock_count > 1)
```

Comment:
This returns false even if lock count is 1. Maybe a different name, such as TRACE_FIRST_LOCK would be more appropriate.

Reply:
updated
@c1728p9